core/controllers/custom_landing_pages.py

```diff
@@ -30,22 +29,26 @@ class FractionLandingRedirectPage(base.BaseHandler):
     @acl_decorators.open_access
     def get(self):
         """Handles GET requests."""
-        self.redirect('/learn/maths/fractions')
+        self.redirect('/math/fractions')
+
+
+class TopicRedirectPage(base.BaseHandler):
+    """The handler redirecting the old landing page URL to the new one."""
+
+    @acl_decorators.open_access
+    def get(self, topic):
+        """Handles GET requests."""
+        self.redirect('/math/%s' % topic)
 
 
 class TopicLandingPage(base.BaseHandler):
     """Page showing the topic landing page."""
 
     @acl_decorators.open_access
-    def get(self, subject, topic):
+    def get(self):
         """Handles GET requests."""
-        if subject in feconf.AVAILABLE_LANDING_PAGES:
-            if topic in feconf.AVAILABLE_LANDING_PAGES[subject]:
-                self.render_template('topic-landing-page.mainpage.html')
-            else:
-                raise self.PageNotFoundException
-        else:
-            raise self.PageNotFoundException
+
+        self.render_template('topic-landing-page.mainpage.html')
```

Comment:
Can we serve this static (through app.yaml)?

Reply:
(I realized that we are generating different URLs for this handler in python, so I don't think we can still do this)